saleor/graphql/order/tests/mutations/test_draft_order_create.py

```diff
@@ -3387,3 +3387,62 @@ def test_draft_order_create_create_no_shipping_method(
     order = Order.objects.get(id=order_pk)
     assert order.undiscounted_base_shipping_price_amount == 0
     assert order.base_shipping_price_amount == 0
+
+
+def test_draft_order_create_voucher_exceed_usage_limit(
+    staff_api_client,
+    permission_group_manage_orders,
+    customer_user,
+    shipping_method,
+    variant,
+    voucher_with_many_codes,
+    channel_USD,
+    graphql_address_data,
+):
+    # given
+    query = DRAFT_ORDER_CREATE_MUTATION
+    permission_group_manage_orders.user_set.add(staff_api_client.user)
+
+    channel_USD.include_draft_order_in_voucher_usage = True
+    channel_USD.save(update_fields=["include_draft_order_in_voucher_usage"])
+
+    voucher = voucher_with_many_codes
+    voucher.usage_limit = 1
+    voucher.save(update_fields=["usage_limit"])
+
+    code_1, code_2, _, _, _ = voucher.codes.all()
+    code_1.used = 1
+    code_1.save(update_fields=["used"])
+
+    user_id = graphene.Node.to_global_id("User", customer_user.id)
+    variant_id = graphene.Node.to_global_id("ProductVariant", variant.id)
+    variant_qty = 2
```

Comment:
We need more extended test.
1) We should have more vouchers
2) Each voucher should have more voucher code with `used` set
3) We should have a test when a limit is not exceeded in the voucher usage (once we should use the code with `used` and the second one we should use the new that was never used)
4) We should have a test when a limit is exceeded in such case
5) We should consider adding e2e test for such scenario - IMO it's worth

Reply:
Tests added. E2E tests will be added in separate PR